src/Controller/LoginController.php

```diff
@@ -0,0 +1,438 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\Module\casserver\Controller;
+
+use SimpleSAML\Auth\ProcessingChain;
+use SimpleSAML\Auth\Simple;
+use SimpleSAML\Configuration;
+use SimpleSAML\HTTP\RunnableResponse;
+use SimpleSAML\Logger;
+use SimpleSAML\Module;
+use SimpleSAML\Module\casserver\Cas\AttributeExtractor;
+use SimpleSAML\Module\casserver\Cas\Factories\ProcessingChainFactory;
+use SimpleSAML\Module\casserver\Cas\Factories\TicketFactory;
+use SimpleSAML\Module\casserver\Cas\Protocol\Cas20;
+use SimpleSAML\Module\casserver\Cas\Protocol\SamlValidateResponder;
+use SimpleSAML\Module\casserver\Cas\ServiceValidator;
+use SimpleSAML\Module\casserver\Cas\Ticket\TicketStore;
+use SimpleSAML\Module\casserver\Controller\Traits\UrlTrait;
+use SimpleSAML\Module\casserver\Http\XmlResponse;
+use SimpleSAML\Session;
+use SimpleSAML\Utils;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\AsController;
+use Symfony\Component\HttpKernel\Attribute\MapQueryParameter;
+
+#[AsController]
+class LoginController
+{
+    use UrlTrait;
```

Comment:
I think this needs the ticket validator trait.

Reply:
Done